src/colorizer/AnnotationData.ts

```diff
@@ -518,6 +524,59 @@ export class AnnotationData implements IAnnotationData {
     return { annotationData, mismatchedTimes, mismatchedTracks, unparseableRows, invalidIds, totalRows: data.length };
   }
 
+  private findMatchingLabelIdx(name: string, type: LabelType): number {
+    for (let i = 0; i < this.labelData.length; i++) {
+      const labelData = this.labelData[i];
+      if (labelData.options.name === name && labelData.options.type === type) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  /**
+   * Merges two annotation data objects and returns a new, resulting object,
+   * based on the merge mode.
+   */
+  static merge(
```

Comment:
Made this a static class method since it needs to have access to private properties of both AnnotationData objects